src/datasets/utils/py_utils.py

```diff
@@ -599,6 +599,109 @@ class Pickler(dill.Pickler):
 
     dispatch = dill._dill.MetaCatchingDict(dill.Pickler.dispatch.copy())
 
+    def save(self, obj, save_persistent_id=True):
+        # lazy registration of reduction functions
+        obj_type = type(obj)
+        if obj_type not in Pickler.dispatch:
+            if (obj_type.__module__, obj_type.__name__) == ("_regex", "Pattern"):
+                try:
+                    import regex
+
+                    if config.DILL_VERSION < version.parse("0.3.6"):
+
+                        @pklregister(obj_type)
+                        def _save_regex(pickler, obj):
+                            dill._dill.log.info(f"Re: {obj}")
+                            args = (
+                                obj.pattern,
+                                obj.flags,
+                            )
+                            pickler.save_reduce(regex.compile, args, obj=obj)
+                            dill._dill.log.info("# Re")
+                            return
+
+                    elif config.DILL_VERSION.release[:3] == version.parse("0.3.6").release:
+
+                        @pklregister(obj_type)
+                        def _save_regex(pickler, obj):
+                            dill._dill.logger.trace(pickler, "Re: %s", obj)
+                            args = (
+                                obj.pattern,
+                                obj.flags,
+                            )
+                            pickler.save_reduce(regex.compile, args, obj=obj)
+                            dill._dill.logger.trace(pickler, "# Re")
+                            return
+
+                except ImportError:
+                    pass
+            elif (obj_type.__module__, obj_type.__name__) == ("torch", "Tensor"):
+                try:
+                    import torch
+
+                    if config.DILL_VERSION < version.parse("0.3.6"):
+
+                        @pklregister(obj_type)
+                        def _save_tensor(pickler, obj):
+                            dill._dill.log.info(f"To: {obj}")
+                            args = (obj.cpu().numpy(),)
```

Comment:
(nit) in case the model is training:
```suggestion
                            if obj.requires_grad:
                                args = (obj.detach().cpu().numpy(),)
                            else:
                                args = (obj.cpu().numpy(),)
```